gatorgrade/generate/generate.py

```diff
@@ -15,6 +15,8 @@ def input_correct(initial_path_list: List[str]) -> Dict:
     # Unify the ending format to avoid different users' different input
     corrected_path = []
     for path in initial_path_list:
+        # Make sure the target path starts from the current directory
+        path = f".{os.path.sep}" + path
```

Comment:
Does this line assume that the current directory is the `search_root`? I'm not sure if it should or shouldn't but if @Yanqiao4396 or @Ochirsaikhan can confirm that this should work in cases where `search_root` isn't `"."` I think that would be best.

Reply:
Hi Seajin, Thanks for reviewing, I'd like to answer this question. The short answer is it wouldn't assume the current directory is search_root. The first thing I'd like to mention is that the paths here are the target paths. As required by @mariakimheinert , the path or file should start from the dictionary users choose. SO THE PATHS HERE ARE NOT ACTUAL ONES!
The reason why the problem won't happen is because of the way I check the target paths in the actual paths. Paths here are string and I checked by using `in` statement. That is why I need concatenate `./` with the target paths. Here, because `../` and `./` share the same part `./`, so the problem won't happen

Reply:
![Screenshot from 2022-05-13 17-47-36](https://user-images.githubusercontent.com/79415648/168394989-feb59159-da6a-49ed-ba01-00c02233ea22.png)
![Screenshot from 2022-05-13 17-47-44](https://user-images.githubusercontent.com/79415648/168395005-2709a9c2-b7d8-41c7-9dfe-dac30245bcd7.png)
Here are two screenshots showing the success under the condition you mentioned

Reply:
I think there still may be an edge case where `search_root` isn't a dotted directory -- for instance, if the user wanted to search through `/home/user/thing` but was in `/home/user/other` when they ran it. I'm not actually sure though so maybe test a situation like that? If that works as well then I'm pretty confident you'll be good and this should work.

Reply:
Thanks for reminding me of this condition. I just made changes and now the minor case you mentioned can be solved correctl
![Screenshot from 2022-05-14 17-34-14](https://user-images.githubusercontent.com/79415648/168448915-8cda8449-2fa1-4c05-90c3-6e93030a5aa5.png)
![Screenshot from 2022-05-14 17-34-18](https://user-images.githubusercontent.com/79415648/168448921-8ea74294-b349-47aa-8f08-188d21877f87.png)
y

Reply:
I don't think that test (with `search_root` being `../docs`) fits the scenario unfortunately; I should have been more clear! I would try testing with a `search_root` that does not have any dots in it, so that you can ensure that the interpretation of the path did not rely on any part being a relative path. I do think that this recent test is a good one though - maybe we should think about adding something like this to the test suite? Definitely a task for later.

Reply:
If I don't get it wrong when you are trying to say one condition where the path without any dot, you mean the scene where the path is absolute. Here this function does have the ability to do it, though the output paths will be a little bit tedious. This will be the last update of this PR from me. Unfortunately, I have to focus on my final (college life is not as easy as I thought🙃), so let's "save" this discussion to the summer. Thanks for your effort and time.
![Screenshot from 2022-05-15 21-47-23](https://user-images.githubusercontent.com/79415648/168507365-ff9cabce-b961-4f2e-95dc-2f86967fcf7d.png)

![Screenshot from 2022-05-15 21-46-24](https://user-images.githubusercontent.com/79415648/168507358-dde60d4b-20de-4c07-b949-4f863ca8e86d.png)

Reply:
Yeah, that's totally fine, @Yanqiao4396! This does verify that everything works as it should, although the paths being tedious is an issue we can fix later (trimming off `search_root` from the start of every path). I think for now this resolves all of my concerns! Thanks for taking the time to verify these issues!